### DISCUSSION TITLE
Data exporter values are empty

### DISCUSSION THREAD

**Discussion by KursatErenColak** [primefaces/community#2595] [PrimeFaces] [answered]
When migrate primefaces v11 to v14 Datatable which has lazy load exporter cannot export the values.

Here is my xhtml; 
![image](https://github.com/user-attachments/assets/98e20976-50ab-4a35-942e-5949555f6b50)

When adding pageOnly property to dataExporter it works fine but without ;
![image](https://github.com/user-attachments/assets/9a7759fe-0439-4fc5-9e05-ece53001a3b4)

 
What's the problem

**Comment by KursatErenColak** [comment]
@melloware  I can't create same scenario with primefaces test project I think it's relate with my setup but it was working with version 11

**Reply by melloware** [reply]
Are you using 14.0.4?  That is what PrimeFaces Test is using.

**Comment by KursatErenColak** [comment]
@melloware Yes it is 14.0.4, I check it  it works Gine but I can't find the problem in my project

**Reply by melloware** [reply]
Hmm without a reproducer showing the issue its impossible for me to say other than it works in the PF Test project.

**Comment by KursatErenColak** [comment]
@melloware Ok, which version of apache poi-ooxml and poi-scratchpad with 14.0.4 I am using 5.2.5

**Reply by melloware** [reply]
yep 5.2.5 is good. that is what I am using.

**Comment by KursatErenColak** [comment]
@melloware I used "exportFunction" in p:columns but it's not trigger when click the export button.

**Reply by melloware** [reply]
can you reproduce with PrimeFaces Test?

**Comment by kurtomerfaruk** [comment, marked as answer]
I think you forgot the ‘rowKey’ field. Can you try adding ‘rowKey’?
